sys/net/gnrc/netif/gnrc_netif.c

```diff
@@ -582,13 +600,30 @@ void gnrc_netif_ipv6_addr_remove(gnrc_netif_t *netif,
                                  const ipv6_addr_t *addr)
 {
     int idx;
+    bool remove_sol_nodes = true;
+    ipv6_addr_t sol_nodes;
 
     assert((netif != NULL) && (addr != NULL));
+    ipv6_addr_set_solicited_nodes(&sol_nodes, addr);
     gnrc_netif_acquire(netif);
-    idx = _addr_idx(netif, addr);
-    if (idx >= 0) {
-        netif->ipv6.addrs_flags[idx] = 0;
-        ipv6_addr_set_unspecified(&netif->ipv6.addrs[idx]);
+    for (unsigned i = 0; i < GNRC_NETIF_IPV6_ADDRS_NUMOF; i++) {
+        if (ipv6_addr_equal(&netif->ipv6.addrs[i], addr)) {
+            netif->ipv6.addrs_flags[idx] = 0;
```

Comment:
I'm getting an unitialized variable error on `idx` here.
```
/home/hydrazine/dev/RIOT/sys/net/gnrc/netif/gnrc_netif.c:611:42: error: ‘idx’ may be used uninitialized in this function [-Werror=maybe-uninitialized]
             netif->ipv6.addrs_flags[idx] = 0;
```

Reply:
Oops. Should be fixed now.